task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -792,6 +793,25 @@ def _handle_trigger_dag_run(
     return _handle_current_task_success(context, ti)
 
 
+def _run_task_state_change_callbacks(
+    task: BaseOperator,
+    kind: Literal[
+        "on_execute_callback",
+        "on_failure_callback",
+        "on_success_callback",
+        "on_retry_callback",
+        "on_skipped_callback",
+    ],
+    context: Context,
+    log: Logger,
+) -> None:
+    for i, callback in enumerate(getattr(task, kind)):
+        try:
+            callback(context)
```

Comment:
Not sure whether we need to check `kind` is valid. As this is an internal thing, it's probably ok to keep it as it is?

Reply:
And we have tests to ensure this.